src/WpBookingsUiModule.php

```diff
@@ -0,0 +1,517 @@
+
+        return $definitions($this->eventManager, $this->eventFactory, $this->_getContainerFactory());
+    }
+
+    /**
+     * Check current screen is screen where EDDBK UI should be rendered.
+     *
+     * @since [*next-version*]
+     *
+     * @return bool Is current page is a page where application should be rendered.
+     */
+    protected function _isOnAppPage()
+    {
+        return in_array(get_current_screen()->id, [
+            $this->bookingsPageId,
+            $this->metaboxPageId,
+        ]);
+    }
+
+    /**
+     * Check current screen is bookings page.
+     *
+     * @since [*next-version*]
+     *
+     * @return bool Is current page is a bookings list page.
+     */
+    protected function _isOnBookingsPage()
+    {
+        return get_current_screen()->id === $this->bookingsPageId;
+    }
+
+    /**
+     * Get app state for booking page.
+     *
+     * @since [*next-version*]
+     *
+     * @param ContainerInterface $c Configuration container of module.
+     *
+     * @return array Front-end application's state on bookings page.
+     */
+    protected function _getBookingsAppState($c)
+    {
+        return [
+            /*
+             * All available statuses in application.
+             */
+            'statuses' => $this->_trigger('eddbk_bookings_translated_statuses', [
+                'statuses' => $this->_getTranslatedStatuses($c->get('booking_logic/statuses'), $c->get('wp_bookings_ui/statuses_labels')),
+            ])->getParam('statuses'),
+
+            /*
+             * Statuses that enabled for filtering bookings.
+             */
+            'screenStatuses' => $this->_trigger('eddbk_bookings_screen_statuses', [
+                'screenStatuses' => $this->_getScreenStatuses($c->get('wp_bookings_ui/screen_options/key'), $c->get('booking_logic/statuses')),
+            ])->getParam('screenStatuses'),
+
+            /*
+             * List of available services.
+             */
+            'services' => $this->_getServices(),
+
+            'statusesEndpoint' => $c->get('wp_bookings_ui/screen_options/endpoint'),
+
+            'endpointsConfig' => $this->_prepareEndpoints($c->get('wp_bookings_ui/endpoints_config')),
+        ];
+    }
+
+    /**
+     * Get all translated statuses.
+     *
+     * @since [*next-version*]
+     *
+     * @param mixed $statuses       List of statuses
+     * @param mixed $statusesLabels Map of statuses and it's labels
+     *
+     * @return array Map of statuse codes and translations.
+     */
+    protected function _getTranslatedStatuses($statuses, $statusesLabels)
+    {
+        $translatedStatuses = [];
+
+        foreach ($statuses as $status) {
+            if ($status === 'none') {
+                continue;
+            }
+            $translatedStatuses[$status] = $this->_containerHas($statusesLabels, $status)
+                ? $this->__($this->_containerGet($statusesLabels, $status))
+                : $status;
```

Comment:
Don't we need to translate the label? I think that was the deal.

Reply:
Hmm... I think that this is exactly what I'm doing here:
```
if (labels has key status code)
  return translate status code's label
else
  return status code
```

Reply:
But if the label is the code, as is the case where there is no known label, then what?